MonoGame.Framework/Input/MouseState.cs

```diff
@@ -137,122 +140,184 @@ public override int GetHashCode()
                 var hashCode = _x;
-                hashCode = (hashCode*397) ^ _y;
-                hashCode = (hashCode*397) ^ _scrollWheelValue;
-                hashCode = (hashCode*397) ^ _horizontalScrollWheelValue;
-                hashCode = (hashCode*397) ^ (int) _leftButton;
-                hashCode = (hashCode*397) ^ (int) _rightButton;
-                hashCode = (hashCode*397) ^ (int) _middleButton;
-                hashCode = (hashCode*397) ^ (int) _xButton1;
-                hashCode = (hashCode*397) ^ (int) _xButton2;
+                hashCode = (hashCode * 397) ^ _y;
+                hashCode = (hashCode * 397) ^ _scrollWheelValue;
+                hashCode = (hashCode * 397) ^ _horizontalScrollWheelValue;
+                hashCode = (hashCode * 397) ^ (int)_buttons;
                 return hashCode;
             }
         }
 
         /// <summary>
         /// Gets horizontal position of the cursor in relation to the window.
         /// </summary>
-		public int X {
-			get {
-				return _x;
-			}
-			internal set {
-				_x = value;
-			}
-		}
+		public int X
+        {
+            get
+            {
+                return _x;
+            }
+            internal set
+            {
+                _x = value;
+            }
+        }
 
         /// <summary>
         /// Gets vertical position of the cursor in relation to the window.
         /// </summary>
-		public int Y {
-			get {
-				return _y;
-			}
-			internal set {
-				_y = value;
-			}
+		public int Y
+        {
+            get
+            {
+                return _y;
+            }
+            internal set
+            {
+                _y = value;
+            }
         }
 
         /// <summary>
         /// Gets cursor position.
         /// </summary>
         public Point Position
         {
-            get { return new Point(_x, _y); }   
+            get { return new Point(_x, _y); }
         }
+
         /// <summary>
         /// Gets state of the left mouse button.
         /// </summary>
-		public ButtonState LeftButton { 
-			get {
-				return _leftButton;
-			}
-			internal set { _leftButton = value; }
-		}
+		public ButtonState LeftButton
+        {
+            get
+            {
+                return ((_buttons & LeftButtonFlag) > 0) ? ButtonState.Pressed : ButtonState.Released;
+            }
+            internal set
+            {
+                if (value == ButtonState.Pressed)
+                {
+                    _buttons = (byte)(_buttons | LeftButtonFlag);
+                }
+                else
+                {
+                    _buttons = (byte)(_buttons & (~LeftButtonFlag));
+                }
+            }
+        }
 
         /// <summary>
         /// Gets state of the middle mouse button.
         /// </summary>
-		public ButtonState MiddleButton { 
-			get {
-				return _middleButton;
-			}
-			internal set { _middleButton = value; }			
-		}
+		public ButtonState MiddleButton
+        {
+            get

```

Comment:
A few of the property declarations are using tabs instead of spaces for indenting.

Reply:
The line endings changed so it marked the line as changed, however it was already using tabs.

To be more precise half of some files are still using tabs, maybe I should just force format the entire file.